core/templates/components/skill-selector/select-skill-modal.component.ts

```diff
@@ -42,6 +42,7 @@ export class SelectSkillModalComponent extends ConfirmOrCancelModal {
   selectedSkillId!: string;
   allowSkillsFromOtherTopics: boolean = false;
   associatedSkillSummaries!: ShortSkillSummary[];
+  skillIdsToExclude: Set<string> = new Set<string>();
```

Comment:
Shouldn't this set be handled such that it could be undefined?

```suggestion
  skillIdsToExclude!: Set<string> = new Set<string>();
```